lib/uplink/bucket_attrs_test.go

```diff
@@ -0,0 +1,189 @@
+			},
+			Volatile: struct {
+				RedundancyScheme storj.RedundancyScheme
+				SegmentsSize     memory.Size
+			}{
+				RedundancyScheme: storj.RedundancyScheme{
+					Algorithm:      storj.ReedSolomon,
+					ShareSize:      memory.KiB.Int32(),
+					RequiredShares: 2,
+					RepairShares:   3,
+					OptimalShares:  4,
+					TotalShares:    5,
+				},
+				SegmentsSize: 688894,
+			},
+		}
+	)
+
+	testPlanetWithLibUplink(t, testConfig{},
+		func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, proj *Project) {
+			before := time.Now()
+			bucket, err := proj.CreateBucket(ctx, bucketName, &inBucketConfig)
+			require.NoError(t, err)
+
+			assert.Equal(t, bucketName, bucket.Name)
+			assert.Falsef(t, bucket.Created.Before(before), "impossible creation time %v", bucket.Created)
+
+			got, err := proj.OpenBucket(ctx, bucketName, &access)
+			require.NoError(t, err)
+			defer ctx.Check(got.Close)
+
+			assert.Equal(t, bucketName, got.Name)
+			assert.Equal(t, inBucketConfig.PathCipher, got.PathCipher)
+			assert.Equal(t, inBucketConfig.EncryptionParameters, got.EncryptionParameters)
+			assert.Equal(t, inBucketConfig.Volatile.RedundancyScheme, got.Volatile.RedundancyScheme)
+			assert.Equal(t, inBucketConfig.Volatile.SegmentsSize, got.Volatile.SegmentsSize)
+
+			err = proj.DeleteBucket(ctx, bucketName)
+			require.NoError(t, err)
+		})
+}
+
+// check that when uploading objects without any specific RS or encryption
+// config, the bucket attributes apply. also when uploading objects _with_ more
+// specific config, the specific config applies and not the bucket attrs.
+func TestBucketAttrsApply(t *testing.T) {
+	var (
+		access         = simpleEncryptionAccess("howdoyouwanttodothis")
+		bucketName     = "dodecahedron"
+		objectPath1    = "vax/vex/vox"
+		objectContents = "Willingham,Ray,Jaffe,Johnson,Riegel,O'Brien,Bailey,Mercer"
+		inBucketConfig = BucketConfig{
+			PathCipher: storj.EncSecretBox,
+			EncryptionParameters: storj.EncryptionParameters{
+				CipherSuite: storj.EncSecretBox,
+				BlockSize:   768,
+			},
+			Volatile: struct {
+				RedundancyScheme storj.RedundancyScheme
+				SegmentsSize     memory.Size
+			}{
+				RedundancyScheme: storj.RedundancyScheme{
+					Algorithm:      storj.ReedSolomon,
+					ShareSize:      (3 * memory.KiB).Int32(),
+					RequiredShares: 3,
+					RepairShares:   4,
+					OptimalShares:  5,
+					TotalShares:    5,
+				},
+				SegmentsSize: 1536,
+			},
+		}
+		testConfig testConfig
+	)
+	// so our test object will not be inlined (otherwise it will lose its RS params)
+	testConfig.uplinkCfg.Volatile.MaxInlineSize = 1
+
+	testPlanetWithLibUplink(t, testConfig,
+		func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet, proj *Project) {
+			_, err := proj.CreateBucket(ctx, bucketName, &inBucketConfig)
+			require.NoError(t, err)
+
+			bucket, err := proj.OpenBucket(ctx, bucketName, &access)
+			require.NoError(t, err)
+			defer ctx.Check(bucket.Close)
+
+			{
```

Comment:
Does this need to be indented? None of the following lines are indented

Reply:
yes, for scope. this scoping ensures that `buf` goes away at the end of the block and that nothing else is using it or depending on it later on.